xilem_core/src/views/adapt.rs

```diff
@@ -89,21 +99,30 @@ where
 ///         })
 ///     )
 /// ```
-pub fn adapt<ParentState, ParentAction, ChildState, ChildAction, Context, ChildView, ProxyFn>(
+pub fn adapt<
+    ParentState,
+    ParentAction,
+    ChildState,
+    ChildAction,
+    Context,
+    ChildView,
+    Message,
+    ProxyFn,
+>(
```

Comment:
Generic params are getting a little bit out of hand, but I guess this is what we signed up for, with a generic xilem_core...

Reply:
Could some of these possibly be "bundled" together under a marker type that implements a trait that only consists of associated types?

Reply:
Sounds interesting, but I'm not exactly sure what you mean, could you show an example (or when you have something more concrete in mind, PR)?

Reply:
I'll try it after this PR is merged, and post a PR if it works out.